pthread_atfork error on Arm processor

Hi,
I finished writing my code that uses nng and I successfully compiled it and run it on my laptop.
Then, I moved the source code to an Odroid-XU4 board, where the code is intended to run, and tried to compiled it there, but I got the following error:

`//usr/local/lib/libnng.a(posix_thread.c.o): In function `nni_plat_init':
posix_thread.c:(.text+0x5a0): undefined reference to `pthread_atfork'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.`

I had no problem in compiling and installing nng on the Odroid-XU4, so I hope that some cmake flag could solve the problem...

The Odroid board is based on an Arm processor and the O.S. installed is Ubuntu 18.04, without any Desktop Environment.

I hope you can help me solve this "undefined reference" issue.
Thank you!

Hi,
I noticed that the same thing happens on Ubuntu 18.04.2 LTS (Desktop, x64 version) as well.
So, it seems that the problem is not related to the ARM architecture but to the specific version of the pthread library that is provided with Ubuntu 18.04.

> /usr/local/lib/libnng.a(posix_thread.c.o): In function `nni_plat_init':
> posix_thread.c:(.text+0x6dc): undefined reference to `pthread_atfork'
> collect2: error: ld returned 1 exit status
> CMakeFiles/droneManager.dir/build.make:302: recipe for target 'droneManager' failed
> make[2]: *** [droneManager] Error 1
> CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/droneManager.dir/all' failed
> make[1]: *** [CMakeFiles/droneManager.dir/all] Error 2
> Makefile:83: recipe for target 'all' failed
> make: *** [all] Error 2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -334,6 +334,7 @@ else ()
 
     nng_check_lib (rt clock_gettime  NNG_HAVE_CLOCK_GETTIME)
     nng_check_lib (pthread sem_wait  NNG_HAVE_SEMAPHORE_PTHREAD)
+    nng_check_lib (pthread pthread_atfork NNG_HAVE_PTHREAD_ATFORK_PTHREAD)
     nng_check_lib (nsl gethostbyname NNG_HAVE_LIBNSL)
     nng_check_lib (socket socket NNG_HAVE_LIBSOCKET)
 

